lib/matplotlib/backend_bases.py

```diff
@@ -2440,6 +2477,19 @@ def stop_event_loop_default(self):
         """
         self._looping = False
 
+    def destroy(self):
+        pass
+
+    @property
+    def manager(self):
+        if self._manager is not None:
+            return self._manager()
+
+    @manager.setter
+    def manager(self, manager):
+        if manager is not None:
+            self._manager = weakref.ref(manager)
```

Comment:
Well you did spot a bug, it meant we couldn't remove the manager... however it hid another bug, now has_attr(canvas, 'manager') will now return True.  I can:
1. change the places where we use hasattr to detect whether the canvas has a manager (will break BC for people using hasattr in the same way)
2. don't set the manager from init if ``None``
3. Treat canvas.manager = None as del canvas.manager

Torn between 1 and 2.